File: alphabit-coding-challenge-03/death_triangle/solutions/death_traingle.py
import math

def distance(x1,y1,x2,y2):
	return math.sqrt((x2-x1)**2+(y2-y1)**2)	

x1,y1 = [int(d) for d in input().split(' ')]
x2,y2 = [int(d) for d in input().split(' ')]
x3,y3 = [int(d) for d in input().split(' ')]

lengths = sorted([distance(x1,y1,x2,y2), distance(x1,y1,x3,y3), distance(x2,y2,x3,y3)])

print(lengths[2]**2 == lengths[1]**2 + lengths[0]**2)

File: alphabit-coding-challenge-03/death_triangle/solutions/death_traingle.pas
program testtri;

function distance (a,c,b,d : double):double;
begin
  distance := Sqrt((b-a)*(b-a)+(d-c)*(d-c));
end;

var
  fx, fy, sx, sy, tx, ty, ab, ac, bc : double;

begin
  readln(fx, fy, sx, sy, tx, ty);

  ab := distance(fx, fy, sx, sy);
  ac := distance(fx, fy, tx, ty);
  bc := distance(sx, sy, tx, ty);

  if (bc*bc = ab*ab+ac*ac) or (ab*ab = ac*ac+bc*bc) or (ac*ac = ab*ab+bc*bc) then
     writeln('True')
  else
     writeln('False');

readln();
end.